Fails building project

I am currently getting the following error while building one of the workspace projects (even though the project is getting built successfully)

```
Errors occurred during the build.
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Attempted to beginRule: MultiRule[], does not match outer scope rule: R/
Errors running builder 'Bndtools Builder' on project 'com.amitinside.sound.provider'.
Build Error!
java.lang.NullPointerException
Build Error!
java.lang.NullPointerException
```

```
org.eclipse.core.runtime.CoreException: Build Error!
	at org.bndtools.builder.BndtoolsBuilder.build(BndtoolsBuilder.java:270)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:833)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.NullPointerException
	at org.bndtools.builder.MarkerSupport.createMarker(MarkerSupport.java:107)
	at org.bndtools.builder.MarkerSupport.createMarkers(MarkerSupport.java:94)
	at org.bndtools.builder.MarkerSupport.setMarkers(MarkerSupport.java:78)
	at org.bndtools.builder.BndtoolsBuilder.report(BndtoolsBuilder.java:283)
	at org.bndtools.builder.BndtoolsBuilder.lambda$build$0(BndtoolsBuilder.java:263)
	at bndtools.central.Central.bndCall(Central.java:665)
	at org.bndtools.builder.BndtoolsBuilder.build(BndtoolsBuilder.java:124)
	... 12 more
```

--- a/bndtools.builder/src/org/bndtools/builder/handlers/component/DSAnnotationErrorHandler.java
+++ b/bndtools.builder/src/org/bndtools/builder/handlers/component/DSAnnotationErrorHandler.java
@@ -44,12 +44,12 @@ public class DSAnnotationErrorHandler extends AbstractBuildErrorDetailsHandler {
             }
         }
 
-        if (md == null) {
+        if (md != null) {
+            result.add(md);
+        } else {
             // No other marker could be created, so add a marker to the bnd file
             result.add(new MarkerData(getDefaultResource(project), attribs, false));
         }
-
-        result.add(md);
 
         return result;
     }

